homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,230 @@
+    else:
+        raise PlatformNotReady("No config data found")
+
+    if hass.data[DOMAIN].get(DISCOVERY_DEVICE):
+        switcher_entity = SwitcherControl(
+            hass, name, icon, hass.data[DOMAIN].get(DISCOVERY_DEVICE))
+    else:
+        raise PlatformNotReady("No device data discoverd")
+
+    async_add_entities([switcher_entity], False)
+
+    return None
+
+
+class SwitcherControl(SwitchDevice):
+    """Home Assistant switch entity."""
+
+    from aioswitcher.devices import SwitcherV2Device
+
+    def __init__(self, hass: HomeAssistant, name: str, icon: Optional[str],
+                 device_data: SwitcherV2Device) -> None:
+        """Initialize the entity."""
+        self._hass = hass
+        self._name = name
+        self._icon = icon
+        self._self_initiated = False
+
+        self._device_data = device_data
+        self._state = device_data.state
+        hass.bus.async_listen(EVENT_SWITCHER_DEVICE_UPDATED,
+                              self.async_update_data)
+
+    @property
+    def device_ip_addr(self) -> str:
+        """Return the device's ip address."""
+        return cast(str, self._device_data.ip_addr)
+
+    @property
+    def electric_current(self) -> float:
+        """Return the electric current."""
+        return cast(float, self._device_data.electric_current)
+
+    @property
+    def remaining_time(self) -> Optional[str]:
+        """Return the remaining time to off command."""
+        return self._device_data.remaining_time \
+            if isinstance(self._device_data.remaining_time, str) \
+            else None
+
+    @property
+    def auto_off_set(self) -> str:
+        """Return the auto off configuration set."""
+        return cast(str, self._device_data.auto_off_set)
+
+    @property
+    def last_data_update(self) -> datetime:
+        """Return the datetime for the last update received by the device."""
+        return cast(datetime, self._device_data.last_data_update)
+
+    @property
+    def last_state_change(self) -> datetime:
+        """Return the datetime for the last state change."""
+        return cast(datetime, self._device_data.last_state_change)
+
+    @property
+    def device_name(self) -> str:
+        """Return the device's name."""
+        return cast(str, self._device_data.name)
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False, entity pushes its state to HA."""
+        return False
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "{}-{}".format(
+            self._device_data.device_id, self._device_data.mac_addr)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        from aioswitcher.consts import STATE_ON as SWITCHER_STATE_ON
+        return cast(bool, self._state == SWITCHER_STATE_ON)
```

Comment:
`return self._state == SWITCHER_STATE_ON`

Reply:
This if for `mypy`'s sake only and doesn't has any effect at runtime.

Reply:
See comment above about `mypy`.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/02af9e9b2a7339faa36e52564a117e0d85d965f8).